bin/eslint-branch.js

```diff
@@ -6,24 +6,33 @@
  * small bit of a big fail, you may get errors that weren't caused by you.
  */
 
-const path = require( 'path' );
 const child_process = require( 'child_process' );
-
-const eslintBin = path.join( '.', 'node_modules', '.bin', 'eslint' );
+const path = require( 'path' );
+const yargs = require( 'yargs' );
 
 const branchName = child_process.execSync( 'git rev-parse --abbrev-ref HEAD' ).toString().trim();
-const rev = child_process
+
+const revision = child_process
 	.execSync( 'git merge-base ' + branchName + ' trunk' )
 	.toString()
 	.trim();
+
 const files = child_process
-	.execSync( 'git diff --name-only ' + rev + '..HEAD' )
+	.execSync( 'git diff --name-only --diff-filter=d ' + revision + ' HEAD' )
```

Comment:
Could you explain these changes? Is there a difference between `rev HEAD` and `rev..HEAD`?

FWIW, I can't replicate the deleted file issue on trunk. 🤔

Reply:
> Could you explain these changes? Is there a difference between rev HEAD and rev..HEAD?

If I read the [documentation](https://git-scm.com/docs/git-diff) correctly, ` rev..HEAD` seems to be a synonym for `rev HEAD`. Given the following warning found in this doc, I thought it would be a good idea to avoid the double dots notation:

> However, "diff" is about comparing two endpoints, not ranges, and the range notations (<commit>..<commit> and <commit>...<commit>) do not mean a range as defined in the "SPECIFYING RANGES" section in gitrevisions[7].

> I can't replicate the deleted file issue on trunk.

That's because this tool is not made to run on `trunk` 🙂. It indeed uses `git merge-base` (see [documentation](https://git-scm.com/docs/git-merge-base)) to find a revision that is common to `trunk` and a given branch (i.e. the point where the branch started diverging).

Reply:
thanks for the clarification!